### DISCUSSION TITLE
useFormContext is null in test suite

### DISCUSSION THREAD

**Discussion by MarkusGnigler** [react-hook-form/react-hook-form#10645] [Q&A] [answered]
### Version Number

7.45.1

### Codesandbox/Expo snack

empty

### Steps to reproduce

I try to extend the testing example from the [docs](https://react-hook-form.com/advanced-usage#TestingForm) with the context api and i got always a null exception on useFormContext hook.

What i'm missing here? Any help are appreciated.

Test suite
```

const mockLogin = vi.fn((email, password) => {
	return Promise.resolve({email, password})
})

describe('', () => {
	it('', async () => {
		//
		render(<App login={mockLogin} />)

		fireEvent.submit(await screen.getByRole('button', {hidden: true}))

		expect(await screen.findAllByRole('alert')).toHaveLength(2)
		expect(mockLogin).not.toBeCalled()
	})
})
```

Test App
```

function App({login}: any) {
	const form = useForm()

	const onSubmit = async (data: any) => {
		await login(data.email, data.password)
		form.reset()
	}

	return (
		<FormProvider {...form}>
			<form onSubmit={form.handleSubmit(onSubmit)} >
				<Email />
				<Password />
				<button type="submit" >SUBMIT</button >
			</form >
		</FormProvider >
	)
}

const Email = () => {
	const {
		register,
		formState: {errors},
	} = useFormContext()

	return (
		<>
			<label htmlFor="email" >email</label >
			<input
				id="email"
				{...register('email', {
					required: 'required',
					pattern: {
						value: /\S+@\S+\.\S+/,
						message: 'Entered value does not match email format',
					},
				})}
				type="email"
			/>
			{errors.email && <span role="alert" >{errors.email.message?.toString() ?? ''}</span >}
		</>
	)
}
const Password = () => {
	const {
		register,
		formState: {errors},
	} = useFormContext()

	return (
		<>
			<label htmlFor="password" >password</label >
			<input
				id="password"
				type="password"
				{...register('password', {
					required: 'required',
					minLength: {
						value: 5,
						message: 'min length is 5',
					},
				})}
			/>
			{errors.password && <span role="alert" >{errors.password?.message.toString() ?? ''}</span >}
		</>
	)
}
```

Used testing libraries are:
"vitest": "^0.32.2"
"@testing-library/jest-dom": "^5.16.5",
"@testing-library/react": "^13.4.0",
"@testing-library/user-event": "^14.4.3",

### Expected behaviour

No null exception on ``useFormContext()``. Same goes for ``useController({name: name as any})``

### Relevant log output

![image](https://github.com/react-hook-form/react-hook-form/assets/82930719/39caf484-7876-444b-a737-42bb9b049139)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by leapful** [comment, marked as answer]
@MarkusGnigler 

Please provide a minimal re-producible GitHub repo or codesandbox url. Only a part of your code implementation does not help to investigate the root cause since it can be caused by test configuration or other mock data handling.

**Reply by MarkusGnigler** [reply]
Thanks for the quick response and pointing me in the right direction. 
It was the project setup. 
On a scratch project it works as expected.

Sorry for the interruption!

**Comment by MarkusGnigler** [comment]
In case anyone has come across the same topic. 
The problem was a mistaken import with `react-hook-form-mui`. 
Just switching to `react-hook-form' import fixes my problem.
